Vector_graphics_on_surfaces/include/CGAL/Vector_graphics_on_surfaces/locally_shortest_path.h

```diff
@@ -0,0 +1,4814 @@
+                    const std::vector<std::array<typename K::Vector_2, 4>>& directions,
+                    const std::vector<std::array<typename K::FT, 4>>& lengths,
+                    const int num_subdiv,
+                    const TriangleMesh &tmesh
+#ifndef CGAL_BSURF_USE_DIJKSTRA_SP
+                    , const Dual_geodesic_solver<typename K::FT>& solver = {}
+#endif
+)
+{
+  namespace PMP = CGAL::Polygon_mesh_processing;
+
+  using FT = typename K::FT;
+
+  std::size_t n=directions.size();
+  std::vector< std::vector<PMP::Face_location<TriangleMesh, typename K::FT>> > result(n);
+
+#ifndef CGAL_BSURF_USE_DIJKSTRA_SP
+  const Dual_geodesic_solver<typename K::FT>* solver_ptr=&solver;
+  Dual_geodesic_solver<typename K::FT> local_solver;
+  if (solver.graph.empty())
+  {
+    solver_ptr = &local_solver;
+    init_geodesic_dual_solver(local_solver, tmesh);
+  }
+#endif
+
+#ifdef CGAL_DEBUG_BSURF
+  std::ofstream debug_cp("/tmp/control_points.xyz");
+  std::ofstream debug_ep("/tmp/end_points.xyz");
+  debug_cp << std::setprecision(17);
+  debug_ep << std::setprecision(17);
+#endif
+  for (std::size_t i=0; i<n; ++i)
+  {
+    Bezier_segment<TriangleMesh, FT> control_loc;
+    for (int k=0;k<4; ++k)
+    {
+      control_loc[k] = straightest_geodesic<K>(center,directions[i][k],lengths[i][k],tmesh).back();
+    }
+
+    #ifdef CGAL_DEBUG_BSURF
+      debug_ep << PMP::construct_point(control_loc[0], tmesh) << "\n";
+      debug_ep << PMP::construct_point(control_loc[3], tmesh) << "\n";
+      debug_cp << PMP::construct_point(control_loc[1], tmesh) << "\n";
+      debug_cp << PMP::construct_point(control_loc[2], tmesh) << "\n";
+    #endif
+
+    std::vector<PMP::Face_location<TriangleMesh, FT>> bezier =
+      recursive_de_Casteljau(tmesh, control_loc, num_subdiv
+#ifndef CGAL_BSURF_USE_DIJKSTRA_SP
+                            , *solver_ptr
+#endif
+                            );
+
+    result[i].reserve(bezier.size());
+    result[i].push_back(bezier[0]);
+    for(std::size_t b=0; b<bezier.size()-1; ++b)
+    {
+      const PMP::Face_location<TriangleMesh, FT>& loc = bezier[b];
+      const PMP::Face_location<TriangleMesh, FT>& loc1 = bezier[b+1];
+
+      // connect the two face locations with shortest path is they are in different faces
+      if (loc.first!=loc1.first)
+      {
+        std::vector<PMP::Edge_location<TriangleMesh, FT>> edge_locations;
+        locally_shortest_path<FT>(loc, loc1, tmesh, edge_locations, solver);
+        result[i].reserve(result[i].size()+edge_locations.size());
+        for (const PMP::Edge_location<TriangleMesh, FT>& e : edge_locations)
+          result[i].push_back(PMP::to_face_location(e, tmesh));
+      }
+      result[i].push_back(loc1);
+    }
+  }
+
+  return result;
+}
+
+
+/*!
+ * \ingroup VGSFunctions
+ * computes a path representing a Bézier polyline (a sequence of Bézier curves having an identical control points,
```

Comment:
Is it just one (what the "an" suggests)?  Then point must be singular.  Or are there two (endpoint/startpoint)?

Reply:
it is a sequence of points. A Bezier is defined by 4 control points, and since Bezier are consecutive, they share a  control point. Only 3 new points for the next Bezier segment + the last point of the previous one. I'll rephrase